hack/e2e-test/run-e2e-test.sh

```diff
@@ -0,0 +1,461 @@
+  echo "Cleaning up Azure infrastructure completed."
+}
+
+# setup_awscli installs the awscli
+function setup_awscli() {
+    if ! $(which aws > /dev/null); then
+      echo "Installing awscli..."
+      if pip3 install --break-system-packages awscli; then
+        echo "Successfully installed awscli."
+      else
+        echo "Failed to install awscli."
+        return 1
+      fi
+    else
+      echo "awscli is already installed."
+    fi
+}
+
+# create_aws_container creates the container for the AWS provider
+function create_aws_container() {
+    echo "Setting up AWS infrastructure..."
+    echo "Creating test bucket..."
+    result=$(aws s3api get-bucket-location --bucket ${TEST_ID} 2>&1 || true)
+    if [[ $result == *NoSuchBucket* ]]; then
+      echo "Creating S3 bucket ${TEST_ID} in region ${AWS_DEFAULT_REGION}"
+      accumulated_exit_code=0
+      aws s3api create-bucket --bucket ${TEST_ID} --region ${AWS_DEFAULT_REGION} --create-bucket-configuration LocationConstraint=${AWS_DEFAULT_REGION} --acl private || accumulated_exit_code=1
+      # Block public access to the S3 bucket
+      aws s3api put-public-access-block --bucket ${TEST_ID} --public-access-block-configuration "BlockPublicAcls=true,IgnorePublicAcls=true,BlockPublicPolicy=true,RestrictPublicBuckets=true" || accumulated_exit_code=1
+      # Deny non-HTTPS requests to the S3 bucket, except for localstack which is exposed on an HTTP endpoint
+      if [[ -z "${LOCALSTACK_HOST:-}" ]]; then
+        aws s3api put-bucket-policy --bucket ${TEST_ID} --policy "{\"Version\":\"2012-10-17\",\"Statement\":[{\"Effect\":\"Deny\",\"Principal\":\"*\",\"Action\":\"s3:*\",\"Resource\":[\"arn:aws:s3:::${TEST_ID}\",\"arn:aws:s3:::${TEST_ID}/*\"],\"Condition\":{\"Bool\":{\"aws:SecureTransport\":\"false\"},\"NumericLessThan\":{\"s3:TlsVersion\":\"1.2\"}}}]}" || accumulated_exit_code=1
+      fi
+      if [ $accumulated_exit_code -ne 0 ]; then
+        echo "Failed to create AWS test bucket."
+        return 1
+      fi
+    else
+      echo $result
+      if [[ $result != *${AWS_DEFAULT_REGION}* ]]; then
+        echo "Bucket already exists in a different region. Please delete the bucket and try again."
+        return 1
+      fi
+    fi
+    echo "Successfully created test bucket."
+    echo "Setting up AWS infrastructure completed."
+}
+
+# usage_aws prints the usage for the AWS provider
+function usage_aws {
+    cat <<EOM
+Usage:
+
+Please make sure the following environment variables are set:
+    AWS_DEFAULT_REGION      Region in which the test bucket is created.
+    LOCALSTACK_HOST         Host of the localstack service. ( required only for testing with localstack)
+EOM
+    return 1
+}
+
+# setup_aws_e2e sets up the AWS infrastructure for the e2e tests including deploying localstack and/or installing the awscli
+function setup_aws_e2e() {    
+    if [[ -n ${LOCALSTACK_HOST:-""} ]]; then
+      make deploy-localstack
```

Comment:
Pass the `KUBECONFIG` available in the `run-e2e-test.sh` environment down to `make deploy-localstack`.